README.md

```diff
@@ -1,634 +1,1828 @@
+time to pay back a loan. These features were not implemented because they still require R&D. It is
+unclear whether these features need to be supported directly by Cardano-Loans or whether, thanks to
+its composability, a separate DeFi protocol will suffice. Using a separate protocol, that "plugs-in"
+to Cardano-Loans may result in less overall complexity than putting all of the features into one
+protocol.
+
+### Linkable Credit Histories
+
+Linking credit histories is very useful for bootstrapping a new identity. In TradFi, entrepreneurs
+can link their personal credit history to their new business' to bootstrap a credit history for the
+business.
+
+This should likely be a separate DeFi protocol. If Bob uses credential A and credential B, he can
+trustlessly link the two by creating a dedicated transaction with a beacon token. This link would be
+easily queryable off-chain, using the beacon token. When a lender wants to query the credit history,
+they would first look up all credentials linked to the target one. Then, they would query all of the
+credentials' credit histories. Nothing would need to change for Cardano-Loans.
+
+### Version Compatibility for Negotiations
+
+Currently, a borrower and a lender need to use the same protocol version to negotiate and enforce a
+loan. There are some technical difficulties with enabling cross-version compatibility:
+
+- If they use different datums, which payment credential would be used?
+- The different versions may not know about each others' beacons. How would proper minting/burning
+of the beacons be enforced?
 
+These questions are currently unanswered which is why this feature still requires R&D.
 
 ## Conclusion
-The Cardano-Loans protocol is a first-attempt at rethinking how the economy of Cardano could evolve. It forgoes reliance on global/external token prices in favor of incentivizing the creation of a CSL-native p2p credit-debt market. It is censorship-resistant, scalable, and straightforward in its design. Wallets and other frontends can integrate the protocol into their UI, and even provide all necessary API query services. Endogenous price & rate discovery empowers users to create their own economy, one that is (at least initially) decoupled from the existing financial system, in pursuit of something more fair, equal, and accessible to all.
 
+The Cardano-Loans protocol is a rethinking of how Cardano's DeFi economy could evolve. It forgoes
+reliance on global/external token prices in favor of incentivizing the creation of a CSL-native p2p
+credit-debt market. It is censorship-resistant, scalable, and straightforward in its design. Wallets
+and other front-ends can easily integrate the protocol into their UI, and even provide all necessary
+API query services. Endogenous price and rate discovery empowers users to create their own economy,
+one that is decoupled from the existing financial system, in pursuit of something more fair, equal,
+and accessible to all.
```

Comment:
```suggestion
API query services. Endogenous price and rate discovery empowers users to create their own economy —
one that is decoupled from the existing financial system — in pursuit of something more fair, equal,
and accessible to all.
```
(these are em dashes) Former version may also be grammatically correct, but considering this is the last sentence & most extraordinary conclusion then I think it helps for it to be punctuated like a speaker was reading it.